tdp/core/variables/service_variables.py

```diff
@@ -180,9 +180,7 @@ def _open_var_file(
     @contextmanager
     def open_var_files(
         self, message: str, paths: list[str], fail_if_does_not_exist: bool = False
-    ) -> Iterator[
-        dict[str, Variables]
-    ]:  # TODO: Transform Dict to OrderedDict with python>3.6
+    ) -> Generator[OrderedDict[str, Variables]]:
```

Comment:
Generator generic takes 3 type parameters: `Generator[YieldType, SendType, ReturnType]`.

```suggestion
    ) -> Generator[OrderedDict[str, Variables], None, None]:
```

Reply:
done